tests/test_matrix_utils_auth_login.py

```
@@ -0,0 +1,1161 @@
+    )
+
+
+import os  # noqa: E402
+
+
+@patch("mmrelay.matrix_utils.save_credentials")
+@patch("mmrelay.matrix_utils.AsyncClient")
+@patch("mmrelay.matrix_utils._create_ssl_context", return_value=None)
+@patch("mmrelay.matrix_utils._normalize_bot_user_id", return_value="@user:matrix.org")
+async def test_login_matrix_bot_api_login_debug_path(
+    _mock_normalize,
```

Ruff (0.15.9)

[warning] Fixture `_mock_normalize` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)